include/boost/geometry/algorithms/detail/buffer/buffer_inserter.hpp

```diff
@@ -175,6 +176,16 @@ struct buffer_range
         }
     }
 
+    static inline bool similar_direction(output_point_type const& p0,
+            output_point_type const& p1,
+            output_point_type const& p2)
+    {
+        typedef arithmetic::general_form<coordinate_type> gf;
```

Comment:
I suggest to change the name because in general context it's not clear general form of what is this. I'd rather explicitly name it `line_2d`, `general_2d_line_form` or something similar. Or even more explicitly `cartesian_line_2d`. Because this is only 2d and only cartesian. It's not really that much "general". ;)

Reply:
Hi @awulkiew @vissarion thanks for all your comments.
It is called general because it implements the so-called "general form" (otherwise it would be "generic").
For the rest I come back to all your remarks, hopefully next Sunday

Reply:
@barendgehrels my point is that it is a "general form" of something specific, i.e. equation of 2d cartesian line. I don't know if "general form" is only used WRT line equations but when I was searching the web I found that this name was always put in quotes so it seems to be rather colloquial term. And I'd assume that many things can have "general form" whatever that may be.

Reply:
Right, sorry for the slight delay. I'm OK with a name change, and with a move.

Note that this class is more useful than just a utility. It implements an infinite line, like we have a point, segment, linestring, and this is a (non OGC) infinite line. Currently in 2d, yes. Over the years there have been questions about this (calculate intersection between two infinite lines) and this code already implements that.

So in that sense it could be moved to the geometry/geometries folder.

But I did not make it yet as generic as all the other stuff, so it is not yet concept based.

Anyway, this is the reason I kept the class very concise and did not put a normal constructor there.

More answers to come. I will also update the description.